test/withGL.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+function withGL(init, fn)
+{
+    if (!PIXI.utils.isWebGLSupported())
+    {
+        return undefined;
+    }
+
+    const renderer = init();
+
+    try
+    {
+        return fn(renderer);
+    }
+    finally
+    {
+        PIXI.utils.destroyTextureCache();
+        renderer.destroy();
```

Comment:
Also, it is possible that I'll add more checks in "finally" later, to check for some wrong renderer states